virt_lightning/shell.py

```diff
@@ -215,87 +187,102 @@ def storage_dir(configuration):
-            "storage_dir",
-        ],
-        help="The action to call.",
+    up_parser.add_argument("--virt-lightning-yaml", **vl_lightning_yaml_args)
+    up_parser.add_argument("--context", **context_args)
+
+    down_parser = action_subparsers.add_parser(
+        "down", help="Destroy a running environment", parents=[parent_parser]
     )
-    parser.add_argument(
-        "--virt-lightning-yaml",
-        default="virt-lightning.yaml",
-        help="point on an alternative virt-lightning.yaml file (default: %(default)s)",
+    down_parser.add_argument("--context", **context_args)
+
+    status_parser = action_subparsers.add_parser(
+        "status",
+        help="List the status of the running environment",
+        parents=[parent_parser],
     )
-    parser.add_argument(
-        "--context",
-        default="default",
-        help="change the name of the context (default: %(default)s)",
+    status_parser.add_argument("--context", **context_args)
+
+    action_subparsers.add_parser(
+        "distro_list", help="List the distribution available", parents=[parent_parser]
     )
-    parser.add_argument(
-        "--config",
-        help="path to configuration file",
-        required=False,
-        default=DEFAULT_CONFIGFILE,
+
+    action_subparsers.add_parser(
+        "storage_dir", help="Print the storage directory", parents=[parent_parser]
     )
 
-    args = parser.parse_args()
+    ansible_inventory_parser = action_subparsers.add_parser(
+        "ansible_inventory",
+        help="Print an ansible_inventory of the running environment",
+        parents=[parent_parser],
+    )
+    ansible_inventory_parser.add_argument("--context", **context_args)
 
-    try:
-        configuration = ReadConfigShell(args.config).load()
-    except FileNotFoundError:
-        print("Configuration file not found")
+    args = main_parser.parse_args()
+    if not args.action:
+        print(usage)
+        print(example)
         exit(1)
-    except Exception as e:
-        print("Can not get configuration from file")
-        raise (e)
-
-    if args.action == "up":
-        up(configuration, args.virt_lightning_yaml, args.context)
-    elif args.action == "down":
-        down(configuration, args.context)
-    elif args.action == "ansible_inventory":
-        ansible_inventory(configuration, args.context)
-    elif args.action == "distro":
-        list_distro(configuration)
-    elif args.action == "clean_all":
-        down(configuration)
-    elif args.action == "status_all":
-        status(configuration)
-    elif args.action == "status":
-        status(configuration, args.context)
-    elif args.action == "status_live":
-        status(configuration, args.context, True)
-    elif args.action == "storage_dir":
-        storage_dir(configuration)
+
+    configuration = Configuration()
+    if args.config:
+        configuration.load_fd(args.config)
+
+    globals()[args.action](configuration=configuration, **vars(args))
```

Comment:
Are you sure about using `globals`?

Reply:
The alternative is to maintain a dict() with a name/function mapping.